xonsh/built_ins.py

```diff
@@ -289,7 +288,7 @@ def _get_runnable_name(fname):
         if os.path.isdir(d):
             files = os.listdir(d)
             if ON_WINDOWS:
-                PATHEXT = builtins.__xonsh_env__.get('PATHEXT')
+                PATHEXT = builtins.__xonsh_env__.get('PATHEXT').split(';')
```

Comment:
This line is the actual fix, rest is just clean up. I can separate those changes if you prefer that.